toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/ConfigPanel.java

```diff
@@ -2,408 +2,169 @@
 
 import java.awt.BorderLayout;
 import java.awt.Color;
-import java.awt.Component;
-import java.awt.Dimension;
 import java.awt.Image;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.awt.event.KeyEvent;
-import java.awt.event.KeyListener;
 import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
-import java.net.HttpURLConnection;
-import java.net.URL;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map.Entry;
 import java.util.Properties;
 
-import javax.imageio.ImageIO;
-import javax.net.ssl.*;
 import javax.swing.BorderFactory;
 import javax.swing.Box;
 import javax.swing.BoxLayout;
 import javax.swing.ImageIcon;
 import javax.swing.JButton;
-import javax.swing.JDialog;
-import javax.swing.JFileChooser;
+import javax.swing.JCheckBox;
+import javax.swing.JFrame;
 import javax.swing.JLabel;
-import javax.swing.JOptionPane;
 import javax.swing.JPanel;
-import javax.swing.JTextField;
+import javax.swing.JTabbedPane;
 
+import com.google.inject.Inject;
+import io.toast.tk.agent.ui.i18n.UIMessages;
+import io.toast.tk.agent.ui.panels.*;
+import io.toast.tk.agent.ui.utils.PanelHelper;
 import org.apache.commons.collections4.EnumerationUtils;
 import org.apache.commons.io.FileUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-/**
- * Stub for displaying Configuration item.
- */
-public class ConfigPanel extends JDialog {
+import io.toast.tk.agent.config.AgentConfigProvider;
+
+
+public class ConfigPanel extends JFrame {
 
 	private static final long serialVersionUID = 1L;
-	
-	private String chromeDriverName = "chromedriver";
-	private String webAppName = "webapp";
-	private String recorderName = "recording";
-	private String apiKeyName = "api";
 
 	private static final Logger LOG = LogManager.getLogger(ConfigPanel.class);
+	
+	private AbstractPanel chromePanel, webAppPanel, recorderPanel, apiKeyPanel, pluginPanel, scriptsPanel,
+			proxyAdressPanel, proxyPortPanel, proxyUserNamePanel, proxyUserPswdPanel;
+	
+	private JButton tryButton;
 
-	private JPanel mainPane, secondPane, chromePanel, webAppPanel, recorderPanel, apiKeyPanel;
+	private JTabbedPane secondPane;
 
-	private JLabel icon;
-	private JTextField textFieldChrome, textFieldWebApp, textFieldRecorder, textFieldApiKey;
-	private JPanel textButtonPanelChrome, textButtonPanelWebApp, textButtonPanelRecorder, textButtonPanelApiKey,
-		iconPanelChrome, iconPanelWebApp, iconPanelRecorder, iconPanelApiKey;
-	private JLabel errorLabelChrome, errorLabelWebApp, errorLabelRecorder, errorLabelApiKey;
-	private JButton fileSearchChrome;
+	public static JCheckBox proxyCheckBox;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Make this "public static proxyCheckBox" field final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1444)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Make proxyCheckBox a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)